osu.Game/Screens/Edit/Timing/MetronomeDisplay.cs

```diff
@@ -248,14 +252,27 @@ protected override void Update()
             if (BeatSyncSource.Clock?.IsRunning != true && isSwinging)
             {
                 swing.ClearTransforms(true);
+                stick.FadeColour(overlayColourProvider.Colour2, 1000, Easing.OutQuint);
+                isSwinging = false;
+
+                // instantly latch if pendulum arm is close enough to center (to prevent awkward delayed playback of latch sound)
+                if (Precision.AlmostEquals(swing.Rotation, 0, 1))
+                {
+                    swing.RotateTo(0);
+                    latch?.Play();
+                    return;
+                }
 
                 using (swing.BeginDelayedSequence(350))
                 {
                     swing.RotateTo(0, 1000, Easing.OutQuint);
-                    stick.FadeColour(overlayColourProvider.Colour2, 1000, Easing.OutQuint);
+                    swing.Delay(250).Schedule(() =>
+                    {
+                        // prevent playing latch sound if metronome has started back up again
+                        if (!isSwinging)
+                            latch?.Play();
+                    });
```

Comment:
Using early-conditional on scheduled delegate isn't enough to prevent double latch sound playback, as you could start swinging, stop swinging, and then start swinging again before the 250ms here passes.

A better way would be to cancel the scheduled delegate once the metronome starts swinging again, i.e.:
```diff
diff --git a/osu.Game/Screens/Edit/Timing/MetronomeDisplay.cs b/osu.Game/Screens/Edit/Timing/MetronomeDisplay.cs
index 33325951e5..d95e3f7d4f 100644
--- a/osu.Game/Screens/Edit/Timing/MetronomeDisplay.cs
+++ b/osu.Game/Screens/Edit/Timing/MetronomeDisplay.cs
@@ -2,6 +2,7 @@
 // See the LICENCE file in the repository root for full licence text.
 
 using System;
+using JetBrains.Annotations;
 using osu.Framework.Allocation;
 using osu.Framework.Audio;
 using osu.Framework.Audio.Sample;
@@ -12,6 +13,7 @@
 using osu.Framework.Graphics.Colour;
 using osu.Framework.Graphics.Containers;
 using osu.Framework.Graphics.Shapes;
+using osu.Framework.Threading;
 using osu.Framework.Timing;
 using osu.Framework.Utils;
 using osu.Game.Beatmaps.ControlPoints;
@@ -37,6 +39,9 @@ public class MetronomeDisplay : BeatSyncedContainer
         private Sample tickDownbeat;
         private Sample latch;
 
+        [CanBeNull]
+        private ScheduledDelegate latchDelegate;
+
         [Resolved]
         private OverlayColourProvider overlayColourProvider { get; set; }
 
@@ -266,12 +271,7 @@ protected override void Update()
                 using (swing.BeginDelayedSequence(350))
                 {
                     swing.RotateTo(0, 1000, Easing.OutQuint);
-                    swing.Delay(250).Schedule(() =>
-                    {
-                        // prevent playing latch sound if metronome has started back up again
-                        if (!isSwinging)
-                            latch?.Play();
-                    });
+                    swing.Delay(250).Schedule(() => latch?.Play(), out latchDelegate);
                 }
             }
         }
@@ -287,6 +287,9 @@ protected override void OnNewBeat(int beatIndex, TimingControlPoint timingPoint,
 
             isSwinging = true;
 
+            latchDelegate?.Cancel();
+            latchDelegate = null;
+
             float currentAngle = swing.Rotation;
             float targetAngle = currentAngle > 0 ? -angle : angle;
 
```

Reply:
Have applied mostly this diff.